src/dmd/dtemplate.d

```diff
@@ -2405,7 +2405,9 @@ void functionResolve(Match* m, Dsymbol dstart, Loc loc, Scope* sc, Objects* tiar
         if (tiargs && tiargs.dim > 0)
             return 0;
 
-        if (fd.semanticRun < PASS.semanticdone)
+        // constructors need a valid scope in order to detect semantic errors
+        if (!fd.isCtorDeclaration &&
+            fd.semanticRun < PASS.semanticdone)
```

Comment:
* Can't other things break now?
* Can't a valid scope be passed in instead?

Reply:
> Can't other things break now 

Destructors and other special funcs like `__postblit` look, at first glance, like some funcs with possible null dereferences of the visitor's scope. But this doesn't mean that this can happen. The same pattern wouldn't work for them for example (must be template).

> Can't a valid scope be passed in instead

I thought `fd._scope`, as this was done a while ago but it's also `null` actually. The semantic visitor is supposed to be created with the scope, so no.